### ISSUE TITLE
Trouble initializing catalogs

### ISSUE THREAD

**Issue by dylanmcreynolds** [bluesky/databroker#645]
When combining databroker==1.2.0 and intake==0.6.1, the following bad behavior can be seen:

```
>>> from intake.catalog import load_combo_catalog
>>> list(load_combo_catalog())
['als']
>>> from databroker import catalog
>>> list(catalog)
[]
```
Intake is finding the catalog I have specified with in a yaml file on disk, but databroker is not.

This is not seen with databroker==1.2.0 and intake==0.6.0

I suspect that this has to do with databroker.discovery.MergedCatalog. It overrides `_make_entries_container() `but not `__iter__()` as suggested in https://github.com/intake/intake/blob/bfa45f7e6c311c3344792a79893fb372eb4e2436/intake/catalog/base.py#L136

**Comment by dylanmcreynolds** [comment]
@gwbischof local testing of your merged fix worked for me. Thanks!

**Comment by danielballan** [comment]
Closed by #647
